public/js/auth.js

```diff
@@ -0,0 +1,162 @@
+async function auth(authType) {
+    if (
+        window.config.backendEndpoint === undefined || 
+        window.config.backendEndpoint === '' ||
+        window.config.projectId === undefined || 
+        window.config.projectId === ''
+    ) {
+        console.error("Missing configuration");
+        return;
+    }
+    let videoClientOptions;
+    if (authType === "auth0") {
+        videoClientOptions = await useAuth0();
+    } else if (authType === "token") {
+        videoClientOptions = await getBroadcasterOptions();
+    } else {
+        console.error("Invalid authType");
+        return;
+    }
+    // Create our VideoClient instance with the options above and once created we can create our Encoder.
+    let vc = await new VideoClient.VideoClient(videoClientOptions);
+    await encoder(vc, VideoClient);
+}
+
+function getUsername() {
```

Comment:
I would move this away to be a utility function, this won't have anything to do with the actual implementation, it is just to create unique usernames for demo purposes.